src/EPR.Payment.Service.Data.UnitTests/Repositories/PaymentsRepositoryTests.cs

```diff
@@ -0,0 +1,199 @@
+﻿using AutoFixture.MSTest;
+using EPR.Payment.Service.Common.Data;
+using EPR.Payment.Service.Common.Data.Repositories;
+using EPR.Payment.Service.Common.UnitTests.Mocks;
+using EPR.Payment.Service.Common.UnitTests.TestHelpers;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Moq.EntityFrameworkCore;
+using System.Data.Entity;
+
+namespace EPR.Payment.Service.Data.UnitTests.Repositories
+{
+    [TestClass]
+    public class PaymentsRepositoryTests
+    {
+        private Mock<DbSet<Common.Data.DataModels.Payment>> _paymentMock;
+        private Mock<DbSet<Common.Data.DataModels.Lookups.PaymentStatus>> _paymentStatusMock;
+        public PaymentsRepositoryTests()
+        {
+            _paymentMock = MockIPaymentRepository.GetPaymentMock();
+            _paymentStatusMock = MockIPaymentRepository.GetPaymentStatusMock(true);
+        }
+
+        [TestMethod]
+        [AutoMoqData]
+        public async Task InsertPaymentStatusAsync_ValidInput_ShouldReturnGuid(
+            [Frozen] Mock<AppDbContext> _dataContextMock, 
+            [Greedy] PaymentsRepository _mockPaymentsRepository)
+        {
+            //Arrange
+            _dataContextMock.Setup(i => i.Payment).ReturnsDbSet(_paymentMock.Object);
+            _mockPaymentsRepository = new PaymentsRepository(_dataContextMock.Object);
+
+            var newId = Guid.NewGuid();
```

Comment:
Freeze the guid values so same value can be reused across the tests and let TestHelper initialize the value for you. This will make code more readable and precise;
        [TestMethod]
        [AutoMoqData]
        public async Task InsertPaymentStatusAsync_ValidInput_ShouldReturnGuid(
            [Frozen] Mock<AppDbContext> _dataContextMock, 
            [Greedy] PaymentsRepository _mockPaymentsRepository,
            [Frozen] Guid newId,
            [Frozen] Guid userId,
            [Frozen] Guid organisationId)
        {
            //Arrange
            _dataContextMock.Setup(i => i.Payment).ReturnsDbSet(_paymentMock.Object);
            _mockPaymentsRepository = new PaymentsRepository(_dataContextMock.Object);

            var request = new Common.Data.DataModels.Payment
            {
                Id = newId,
                UserId = userId,
                OrganisationId = organisationId,
            };

Reply:
resolved